lib/mcwamp.cpp

```diff
@@ -329,19 +329,25 @@ static void read_code_bundles(std::vector<_code_bundle>& bundles) {
   }
 }
 
-void LoadInMemoryProgram(KalmarQueue* pQueue) {
+
+
+void LoadInMemoryProgram(KalmarDevice* k_device) {
 
   static std::vector<_code_bundle> bundles;
   static std::once_flag f;
```

Comment:
`std::once_flag` may be broken in case:

- applications somehow link with HCC runtime statically
- applications somehow loads 2 different versions of HCC runtime

Reply:
please refer to examples in SWDEV-178515 for reduced cases to break `std::once_flag` and possible ways to fix it with FIFO.

Reply:
Somehow loading 2 different versions of the HCC runtime is a bug, and you have more problems than this. We don't provide a static library so it's unclear exactly how applications "somehow" statically link in a supported way. We'd need to actually provide an archive / static version of the HCC RT, which is not yet provided. So this should be dealt with there, since it's in and of itself a different kettle of fish. This has nothing to do with `once_flag` and everything to do with how internal linkage works. You could use and `atomic_flag`, or just a static `bool`, and the challenges for those two unsupported scenarios would be exactly the same.

Reply:
Fixing potential issues exposed by a static version of HCC RT (we don't have one) is going to be future work unrelated to this PR.